src/NativeScript/ObjC/Constructor/ObjCConstructorBase.mm

```diff
@@ -146,8 +148,95 @@ static void writeAdapter(ExecState* execState, const JSValue& value, void* buffe
     visitor.append(constructor->_initializers.begin(), constructor->_initializers.end());
 }
 
+static WTF::String computeInitializerJSName(ExecState* execState, JSFinalObject* initializer, MarkedArgumentBuffer& arguments) {
+    PropertyNameArray properties(execState, PropertyNameMode::Strings);
+    initializer->getOwnPropertyNames(initializer, execState, properties, EnumerationMode(DontEnumPropertiesMode::Exclude, JSObjectPropertiesMode::Include));
+
+    VM& vm = execState->vm();
+    WTF::StringBuilder builder;
+    builder.reserveCapacity(32);
+    for (auto& property : properties) {
+        const WTF::String& propertyString = property.string();
+        const WTF::String firstCharacter = propertyString.substringSharingImpl(0, 1).upper();
+        const WTF::String rest = propertyString.substringSharingImpl(1);
+
+        builder.append(firstCharacter);
+        builder.append(rest);
+        arguments.append(initializer->getDirect(vm, property));
+    }
+
+    return builder.toString();
+}
+
+static JSValue getInitializerForSwiftStyleConstruction(ExecState* execState, JSValue prototype, WTF::String jsName) {
+    static WTF::String initWith("initWith", WTF::String::ConstructFromLiteral);
+    static WTF::String init("init", WTF::String::ConstructFromLiteral);
+    static WTF::String error("Error", WTF::String::ConstructFromLiteral);
+
+    JSValue value;
+#define RETURN_IF_CELL if (value.isCell()) return value
+    
+    if (jsName.isEmpty()) {
+        return value;
+    }
+
+    value = prototype.get(execState, Identifier::fromString(execState, WTF::String(initWith + jsName)));
+    RETURN_IF_CELL;
+
+    value = prototype.get(execState, Identifier::fromString(execState, WTF::String(initWith + jsName + error)));
+    RETURN_IF_CELL;
+
+    value = prototype.get(execState, Identifier::fromString(execState, WTF::String(init + jsName)));
+    RETURN_IF_CELL;
+
+    value = prototype.get(execState, Identifier::fromString(execState, WTF::String(init + jsName + error)));
+    RETURN_IF_CELL;
+
+    return JSValue();
+}
+
 static EncodedJSValue JSC_HOST_CALL constructObjCClass(ExecState* execState) {
     ObjCConstructorBase* constructor = jsCast<ObjCConstructorBase*>(execState->callee());
+
+    if (execState->argumentCount() <= 1) {
+        MarkedArgumentBuffer initializerArguments;
+        JSValue initializer;
+        if (JSFinalObject* argument = jsDynamicCast<JSFinalObject*>(execState->argument(0))) {
+            initializer = getInitializerForSwiftStyleConstruction(execState, constructor->instancesStructure()->storedPrototype(), computeInitializerJSName(execState, argument, initializerArguments));
+        } else if (execState->argumentCount() == 0) {
+            initializer = constructor->instancesStructure()->storedPrototype().get(execState, Identifier::fromString(execState, "init"));
+        } else {
+            goto new_legacy;
+        }
+
+        if (initializer && initializer.isCell()) {
```

Comment:
You could conceivably invoke an initializer that accepts a single `NSDictionary` parameter and feed it an object literal using the legacy constructor resolution logic. I admit it's far-fetched, buuuuuut...